README.md

```diff
@@ -32,84 +34,229 @@ The red balls represent the position of the drones.
+### Numpy version
+
+Basic version which can be used for training, simulation and the real drones.
+
+Execution :
+```python
+from crazy_rl.multi_agent.numpy.circle.circle import Circle
+
+env: ParallelEnv = Circle(
+    drone_ids=np.array([0, 1]),
+    render_mode="human",    # or real, or None
+    init_flying_pos=np.array([[0, 0, 1], [2, 2, 1]]),
+)
+
+done = False
+while not done:
+    # Execute policy for each agent
+    actions: Dict[str, np.ndarray] = {}
+    for agent_id in env.possible_agents:
+        obs_with_id = torch.Tensor(concat_id(obs[agent_id], agent_id)).to(device)
+        act, _, _ = actor.get_action(obs_with_id.unsqueeze(0)) # YOUR POLICY HERE
+        act = act.detach().cpu().numpy()
+        actions[agent_id] = act.flatten()
+
+    obs, _, _, _, _ = env.step(actions)
+```
+
+You can have a look at the [test_multiagent](learning/exec_multiagent.py) file. The path to the save model MASAC and the mode to "real" has to be set on the main.
+
+### JAX version
+
+This version is specifically optimized for GPU usage and intended for agent training purposes.
+However, simulation and real-world functionalities are not available in this version.
+
+Execution:
+
 ```python
-    env: ParallelEnv = Circle(
-        drone_ids=np.array([0, 1]),
-        render_mode="human",    # or real, or None
-        init_flying_pos=np.array([[0, 0, 1], [2, 2, 1]]),
+from jax import random
+from crazy_rl.multi_agent.jax.circle.circle import Circle
+
+parallel_env = Circle(
+        num_drones=5,
+        init_flying_pos=jnp.array([[0.0, 0.0, 1.0], [2.0, 1.0, 1.0], [0.0, 1.0, 1.0], [2.0, 2.0, 1.0], [1.0, 0.0, 1.0]]),
+        num_intermediate_points=100,
     )
 
-    done = False
-    while not done:
-        # Execute policy for each agent
-        actions: Dict[str, np.ndarray] = {}
-        for agent_id in env.possible_agents:
-            obs_with_id = torch.Tensor(concat_id(obs[agent_id], agent_id)).to(device)
-            act, _, _ = actor.get_action(obs_with_id.unsqueeze(0)) # YOUR POLICY HERE
-            act = act.detach().cpu().numpy()
-            actions[agent_id] = act.flatten()
-
-        obs, _, _, _, _ = env.step(actions)
+seed = 5  # test value
+key = random.PRNGKey(seed)
+
+key, subkey = random.split(key)
+state = parallel_env.reset(subkey)
+
+for i in range(500):
+    while not (jnp.any(state.terminations) or jnp.any(state.truncations)):
+
+        # where you would choose the drones' actions (2D array)
+
+        key, subkey = random.split(key)
+        state = parallel_env.step(state, actions, key)
+
+        # where you would learn or add to buffer
+
+    key, subkey = random.split(key)
+    state = parallel_env.reset(subkey)
 ```
 
-You can have a look at the [test_multiagent](learning/test_multiagent.py) file. The path to the save model MASAC and the mode to "real" has to be set on the main.
+### Vmapped JAX version
+
+The JAX version supports vmap, enabling parallelized training, particularly beneficial for GPU utilization.
```

Comment:
The JAX version supports vectorized operations such as `vmap`, enabling parallelized training, allowing to leverage all the cores on the GPU.